core/util/envinfo.go

```diff
@@ -0,0 +1,103 @@
+/*
+|    Protect your secrets, protect your sensitive data.
+:    Explore VMware Secrets Manager docs at https://vsecm.com/
+</
+<>/  keep your secrets… secret
+>/
+<>/' Copyright 2023–present VMware, Inc.
+>/'  SPDX-License-Identifier: BSD-2-Clause
+*/
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/vmware-tanzu/secrets-manager/core/log"
+)
+
+// PrintEnvironmentInfo prints information about specific environment variables, enabled features, and binary version.
+func PrintEnvironmentInfo(id *string, envVarsToPrint []string) {
```

Comment:
I have mixed feelings about keeping this in Util versus moving it into the logging package. — Let’s keep it this way; but don’t be surprised if I refactor it elsewhere :) .